Index: src/personajes/Alien.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>package personajes;\r\n\r\npublic class Alien extends Jugador{\r\n\r\n    public Alien(String nombre, int ataque, int defensa, int vidas) {\r\n        super(nombre, ataque, defensa, vidas);\r\n\r\n        System.out.println(\"Soy el constructor de Alien pero estoy creando un \" + this.getClass().getSimpleName());\r\n    }\r\n\r\n    public String toString() {\r\n        return super.toString();\r\n    }\r\n\r\n    @Override\r\n    protected void setDefensa(int defensa) {\r\n        super.setDefensa(defensa);\r\n    }\r\n\r\n    @Override\r\n    protected void setVidas(int vidas) {\r\n        if (vidas > 20){\r\n            super.setVidas(vidas + 3);\r\n        } else super.setVidas(vidas);\r\n\r\n    }\r\n\r\n    @Override\r\n    public int getDefensa() {\r\n        return super.getDefensa();\r\n    }\r\n\r\n    @Override\r\n    public int getVidas() {\r\n        return super.getVidas();\r\n    }\r\n}\r\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/personajes/Alien.java b/src/personajes/Alien.java
--- a/src/personajes/Alien.java	(revision a3ff58f69337d60095a1aceb282aee35551ea78f)
+++ b/src/personajes/Alien.java	(date 1742578982751)
@@ -34,4 +34,29 @@
     public int getVidas() {
         return super.getVidas();
     }
+
+    @Override
+    public void ataca(Jugador jugador) {
+        System.out.println("\n===== INICIO DEL COMBATE =====");
+        System.out.println("ANTES DEL ATAQUE:");
+        System.out.println("\tAtacante: " + this);
+        System.out.println("\tAtacado: " + jugador);
+
+        if (this.getVidas() > 20){
+            System.out.println("\n⚠️ " + this.getNombre() + " enloquece! (+3 ATAQUE, -3 DEFENSA)");
+            this.setAtaque(getAtaque() + 3);
+            this.setDefensa(getDefensa() - 3);
+        }
+
+        System.out.println("\n>>> " + this.getNombre() + " ataca a " + jugador.getNombre());
+        jugador.esGolpeadoCon(this.getAtaque());
+
+        System.out.println("\n>>> " + jugador.getNombre() + " contraataca " + this.getNombre());
+        this.esGolpeadoCon(jugador.getAtaque());
+
+        System.out.println("\nDESPUÉS DEL ATAQUE:");
+        System.out.println("\tAtacante: " + this);
+        System.out.println("\tAtacado: " + jugador);
+        System.out.println("===== FIN DEL COMBATE =====");
+    }
 }
